### PR TITLE
Update latest dependencies, closes #1

### LINKED ISSUE #1: Jitpack报错： No read access
可能是由于错误配置或者登录失效等原因？

See: https://jitpack.io/#Fallen-Breath/caonditional-mixin

### PR DESCRIPTION
**Pull request by SteveKunG**
This is hopefully fixes compiling issue with Forge/NeoForge 1.20.6+.
Note: I haven't tested in production yet.

### CODE REVIEW

**Review by Fallen-Breath — CHANGES_REQUESTED** [review]
Your changes breaks the compatibility for non java 21 runtime

**Comment by SteveKunG** [comment]
Done! Is there anything else?

**Comment by Fallen-Breath** [comment]
Please test your builds in production, espeically in older non-java-21 versions (e.g. mc 1.16, 1.18, 1.19)

**Comment by SteveKunG** [comment]
Tested all version work fine.
Fabric/Forge 1.16.5, 1.18.2, 1.19.2, 1.20.1, 1.20.6, 1.21.1
NeoForge 1.20.6 and 1.21.1

Note: Forge below 1.20.1 when viewing conditional mixin in the mod menu will crash the game due to 
`java.lang.IllegalArgumentException: Root resources can only be filenames, not paths (no / allowed!)`
https://github.com/Fallen-Breath/conditional-mixin/blob/master/forge/src/main/resources/META-INF/mods.toml#L11

File: forge/build.gradle [hunk 1]
```diff
@@ -22,6 +22,10 @@ dependencies {
 	shadowCommon(project(path: ":common", configuration: "transformProductionForge")) { transitive = false }
 }
 
+configurations.configureEach {
```
Comment (Fallen-Breath) [inline forge/build.gradle:25]: what is this change for?

Comment (SteveKunG) [reply forge/build.gradle:25]: A requirement for Forge version 50+ to able run on dev. (Still I can't get it to run on my machine)

I took from Architectury discord.
https://discord.com/channels/792699517631594506/792701725106634783/1272848116864909314

**Pull request merged**